.github/workflows/linter.yml

```diff
@@ -24,6 +24,13 @@ jobs:
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0 # If you use VALIDATE_ALL_CODEBASE = true, you can remove this line to improve performances
+      - name: Set environment for branch
+        run: |
+          if [[ $GITHUB_REF_NAME == 'master' ]]; then
+              echo "VALIDATE_ALL_CODEBASE: true" >> "$GITHUB_ENV"
+          else
+              echo "VALIDATE_ALL_CODEBASE: false" >> "$GITHUB_ENV"
+          fi
```

Comment:
We might want to validate the full codebase on any feature branch also, not just master (not that we have many feature branches yet).  However, specifically on events that were triggered from PRs, we only want the diff...